File: BankApplicationAPI/BankApplicationAPI/Models/LoanApplication.cs
namespace BankApplicationAPI.Models;

public partial class LoanApplication
{
    public int LoanId { get; set; }

    public string? CustomerId { get; set; }

    public int? LoanTypeId { get; set; }

    public decimal? LoanAmount { get; set; }

    public DateTime? ApplicationDate { get; set; }

    public string? Files { get; set; }

    public string? LoanStatus { get; set; }

    public string? EmployeeId { get; set; }

    public DateTime? ApprovalDate { get; set; }

    public string? Comments { get; set; }

    public virtual Customer? Customer { get; set; }

    public virtual Employee? Employee { get; set; }

    public virtual ICollection<LoanPaymentSchedule> LoanPaymentSchedules { get; set; } = new List<LoanPaymentSchedule>();

    public virtual ICollection<LoanRepaymentLog> LoanRepaymentLogs { get; set; } = new List<LoanRepaymentLog>();

    public virtual LoanType? LoanType { get; set; }
}

File: BankApp/src/app/Modules/LoanApplication.ts
import { Customer } from "./Customer";
import { Employee } from "./Employee";
import { LoanPaymentSchedule } from "./LoanPaymentSchedule";
import { LoanRepaymentLog } from "./LoanRepaymentLog";
import { LoanType } from "./LoanType";

export interface LoanApplication {
    loanId?: number | null;
    customerId?: string | null;
    loanTypeId?: number | null;
    loanAmount?: number | null;
    applicationDate?: Date | null;
    files?: string | null;
    loanStatus?: string | null;
    employeeId?: string | null;
    approvalDate?: Date | null;
    comments?: string | null;
    customer?: Customer | null;
    employee?: Employee | null;
    loanPaymentSchedules?: LoanPaymentSchedule[];
    loanRepaymentLogs?: LoanRepaymentLog[];
    loanType?: LoanType | null;
}